test/form_builder/wrapper_test.rb

```diff
@@ -59,6 +59,15 @@ class WrapperTest < ActionView::TestCase
     assert_no_select 'input.is-invalid'
   end
 
+  test 'wrapper does not determine if valid class is needed when it is set to nil' do
```

Comment:
This test is not covering your changes - I put back the deleted code and it still pass. One way I can think right in how to test it, would be with something like:

```ruby
  test 'wrapper does not determine if valid class is needed when it is set to nil' do
    @user.instance_eval { undef errors }
    with_form_for @user, :name, wrapper: custom_wrapper_with_input_valid_class(valid_class: nil)

    assert_no_select 'div.field_without_errors'
  end
```
A better test would checking if `input.valid?` has not been called at all, but I don't know how to do it :see_no_evil: